node/network/availability-recovery/src/lib.rs

```diff
@@ -686,6 +687,38 @@ impl Future for RecoveryHandle {
 	}
 }
 
+/// Cached result of an availability recovery operation.
+#[derive(Debug, Clone)]
+enum CachedValidity {
```

Comment:
isn't this rather a `AvailabilityRecoveryStatus`? Skip the `Availability` if it's too long, but `CachedValidity` is not indicating what it represents.

Reply:
I ended up with `CachedRecovery` as it is concise and describes what it is the best.